src/vs/workbench/common/editor.ts

```diff
@@ -612,6 +633,14 @@ export interface IEditorInput extends IDisposable {
 	isDisposed(): boolean;
 }
 
+export function isIEditorInput(editor: unknown): editor is IEditorInput {
```

Comment:
You wrote this function but we also have https://github.com/microsoft/vscode/blob/344c11408bd744af2001f0273e42b040d4637f4a/src/vs/workbench/common/editor/editorInput.ts#L168-L170. Do we ever have an IEditorInput that isn't an instance of `EditorInput` I've been using that to discern from untyped and it seems to return if `is IEditorInput` instead of `is EditorInput`

Reply:
Yeah you are right, there is no `IEditorInut` that is not `EditorInput`. But due to cyclic dependency issues, I cannot depend on `editorInput.ts` from `editor.ts`. If you have a better idea let me know, maybe I should have not moved `editorInput.ts` into its own file...

Reply:
Well can't we just delete the function in `editorInput.ts` and use the `editor.ts` version everywhere?

Reply:
Yeah we could but I am worried that the `isIEditorInput` method might perform worse compared to `instanceof` check and I wanted to keep this PR small as possible.